src/app/clusters/operational-state-server/operational-state-delegate.h

```diff
@@ -0,0 +1,393 @@
+     * @return void
+     */
+    virtual void ReleaseOperationalStateList(GenericOperationalStateList * operationalStateList) = 0;
+
+    /**
+     * Get operational phase list.
+     * @param operationalPhaseList The pointer to operational phase list.
+     * After a successful return the caller is responsible for calling ReleaseOperationalPhaseList on the outparam.
+     */
+    virtual CHIP_ERROR GetOperationalPhaseList(GenericOperationalPhaseList ** operationalPhaseList, size_t & size) = 0;
+
+    /**
+     * Release operational phase list
+     * @param operationalStateList The pointer for which to clear the GenericOperationalPhaseList.
+     * @return void
+     */
+    virtual void ReleaseOperationalPhaseList(GenericOperationalPhaseList * operationalPhaseList) = 0;
+
+    /**
+     * Get operational error.
+     * @param void.
+     * @return the const reference of operational error.
+     */
+    virtual const GenericOperationalError GetOperationalError() const = 0;
+
+    /**
+     * Set operational state.
+     * @param opState The operational state for which to set.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    virtual CHIP_ERROR SetOperationalState(const GenericOperationalState & opState) = 0;
+
+    /**
+     * Set operational error.
+     * @param opErrState The reference of operational error.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    virtual CHIP_ERROR SetOperationalError(const GenericOperationalError & opErrState) = 0;
+
+    // command callback
+    /**
+     * Handle Command Callback in application: Pause
+     * @param[out] get operational error after callback.
+     * @return operational error after callback
+     */
+    virtual GenericOperationalError & HandlePauseStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Handle Command Callback in application: Resume
+     * @param[out] get operational error after callback.
+     * @return operational error after callback
+     */
+    virtual GenericOperationalError & HandleResumeStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Handle Command Callback in application: Start
+     * @param[out] get operational error after callback.
+     * @return operational error after callback
+     */
+    virtual GenericOperationalError & HandleStartStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Handle Command Callback in application: Stop
+     * @param[out] get operational error after callback.
+     * @return operational error after callback
+     */
+    virtual GenericOperationalError & HandleStopStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Send OperationalError Event
+     * @param[in] set the operational error to event.
+     * @return true: send event success; fail : send event fail.
+     */
+    virtual bool sendOperationalErrorEvent(const GenericOperationalError & err) = 0;
```

Comment:
Why would the _delegate_ be sending the event?  I would expect the SDK to handle that...  Or is this not about events in the spec sense at all?

Also, why the lower-cased naming?